selection.go

```diff
@@ -203,8 +206,13 @@ func (s *controlledSelector) ContactCandidates() {
 			s.agent.checkKeepalive()
 		}
 	} else {
-		s.log.Trace("pinging all candidates")
-		s.agent.pingAllCandidates()
+		if time.Since(s.startTime) > s.agent.connectionTimeout {
```

Comment:
I think we should use s.agent.candidateSelectionTimeout instead

other than that, LGTM!

Reply:
Pushed that change. :+1: